src/Materials/Textures/cubeTexture.ts

```diff
@@ -294,13 +294,44 @@ export class CubeTexture extends BaseTexture {
             this.name = url;
         }
         this.url = url;
-        this.delayLoadState = Constants.DELAYLOADSTATE_NOTLOADED;
-        this._prefiltered = prefiltered;
-        if (this._prefiltered) {
+
+        const lastDot = url.lastIndexOf(".");
+        const extension = forcedExtension ? forcedExtension : (lastDot > -1 ? url.substring(lastDot).toLowerCase() : "");
+        const isDDS = (extension.indexOf(".dds") === 0);
+        const isEnv = (extension.indexOf(".env") === 0);
+
+        if (isEnv) {
             this.gammaSpace = false;
+            this._prefiltered = false;
             this.anisotropicFilteringLevel = 1;
         }
-        this._forcedExtension = forcedExtension || null;
+        else {
+            this._prefiltered = prefiltered;
+
+            if (prefiltered) {
+                this.gammaSpace = false;
+                this.anisotropicFilteringLevel = 1;
+            }
+        }
+        if (this._files) {
+            this._files.length = 0;
+        } else {
+            this._files = [];
+        }
+
+        if (!isEnv && !isDDS && !extensions) {
+            extensions = ["_px.jpg", "_py.jpg", "_pz.jpg", "_nx.jpg", "_ny.jpg", "_nz.jpg"];
+        }
+
+        if (extensions) {
+
+            for (var index = 0; index < extensions.length; index++) {
+                this._files.push(url + extensions[index]);
+            }
+            this._extensions = extensions;
+        }
+
+        this.delayLoadState = Constants.DELAYLOADSTATE_NOTLOADED;

```

Comment:
Maybe we should factorize this code as most of it is used in the constructor?

Reply:
sure, i'll work on that

Reply:
Was refactored. the cubetexture constructor now uses updateURL to set the texture